File: Javascript/Benchmark.js
class Benchmark {
    constructor(iterations = 10000) {
        this.iterations = iterations;
        this.results = {};
    }

    measure(fn, params = []) {
        const start = performance.now();

        for (let i = 0; i < this.iterations; i++) {
            fn(...params);
        }

        const end = performance.now();
        const elapsed = (end - start) / 1000; // Convert to seconds

        this.results = {
            time: elapsed.toFixed(10),
            iterations: this.iterations,
            time_per_iteration: (elapsed / this.iterations).toFixed(10)
        };
        
        return this.results;
    }

    print() {
        console.log(`${this.results.time} s`);
    }
}

module.exports = Benchmark;

File: Rust/Benchmark.rs
use std::time::Instant;

pub struct Benchmark {
    iterations: usize,
    time: f64,
    time_per_iteration: f64,
}

impl Benchmark {
    pub fn new(iterations: usize) -> Self {
        Self {
            iterations,
            time: 0.0,
            time_per_iteration: 0.0,
        }
    }

    pub fn measure<F>(&mut self, mut func: F)
    where
        F: FnMut(),
    {
        let start = Instant::now();

        for _ in 0..self.iterations {
            func();
        }

        let duration = start.elapsed();
        self.time = duration.as_secs_f64();
        self.time_per_iteration = self.time / self.iterations as f64;
    }

    pub fn print(&self) {
        println!("{:.10} s", self.time);
    }
}
